icon.schema.json

```diff
@@ -21,7 +21,32 @@
     "aliases": {
       "type": "array",
       "items": {
-        "type": "string"
+        "oneOf": [
+          {
+            "type": "string"
+          },
+          {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string"
+              },
+              "deprecated": {
+                "const": true
+              },
+              "deprecationReason": {
+                "$ref": "#/$defs/deprecationReasons"
+              },
+              "gracePeriod": {
```

Comment:
I asked copilot for a better name since I had no clue from just looking at the variable name what it is gonna do.
```suggestion
              "toBeRemovedInVersion": {
```

Reply:
Another option could be a boolean `toBeRemovedInNextMajorVersion` 🤷

Reply:
Fine by me, renamed it.